### ISSUE TITLE
Routing when Admin base URL differs

### ISSUE THREAD

**Issue by danelowe** [CodistoConnect/CodistoConnect#238]
When the admin and main website are on different base URLs, `Mage_Core_Controller_Varien_Front::_checkBaseUrl()` forces any /codisto/ URL to be redirected to the store base URL. 

This is because `Mage_Core_Controller_Varien_Front::_isAdminFrontNameMatched()` returns false. This is in turn because there is no admin router path registered with `<frontName>codisto</frontName>`

**Comment by lamery** [comment]
Hi Dane,

I'm struggling a little with this one, are you suggesting we just need to add an admin frontname section to the plugin's config.xml?

Perhaps you could describe the environment so we can test it for ourselves.

It is just one magento server instance, admin base url is set to one domain, main website base url is set to another completely different domain?

If you can confirm that is the setup, I'll sort out a test environment so we can ensure we resolve this issue.

**Comment by danelowe** [comment]
Hi Luke,

Yes, this is on one Magento server instance where admin base URL is set to one domain, and the main website base URL is set to another completely different domain. 

I actually discovered this in dev where:

- Default config Base URL is set to http://localhost:9292/
- Main Website config Base URL is set to http://danes-macbook-pro.local:9292/ (So I could test front-end on virtual machines / physical phone)

Because `/codisto/` was not recognised as an admin URL, and the request path did not match the front-end base URL `_checkBaseUrl` was redirecting to the home page before the custom router was even called.

Yes, I did get it working by adding the following to `config.xml`, although I don't know enough about the extension to know if that is the best solution.

```
	<admin>
		<routers>
			<codisto-sync>
				<use>admin</use>
				<args>
					<module>Codisto_Sync</module>
					<frontName>codisto</frontName>
				</args>
			</codisto-sync>
			...
		</routers>
	</admin>
```
